zoom_rooms/indico_zoom_rooms/util.py

```diff
@@ -0,0 +1,88 @@
+# This file is part of the CERN Indico plugins.
+# Copyright (C) 2024 CERN
+#
+# The CERN Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License; see
+# the LICENSE file for more details.
+
+import typing as t
+from itertools import chain
+
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.models.events import Event
+from indico.modules.events.sessions.models.blocks import SessionBlock
+from indico.modules.events.sessions.models.sessions import Session
+from indico.modules.rb.models.rooms import Room
+from indico.modules.vc.models.vc_rooms import VCRoom, VCRoomEventAssociation
+
+
+def get_zoom_room_id(obj: Event | Contribution | SessionBlock | Room) -> str | None:
+    """Get the Zoom Room Exchange user id for an object (e.g. event) or room.
+
+    Returns `None` if there isn't one.
+    """
+    room = obj if isinstance(obj, Room) else obj.room
+
+    if room:
+        return room.get_attribute_value('zoom-rooms-calendar-id')
+    else:
+        return None
+
+
+def make_zoom_room_entry_id(user_id: str, obj: Event | Contribution | SessionBlock, vc_room: VCRoom) -> str:
+    event_id = obj.id if isinstance(obj, Event) else obj.event.id
+    sub_id = ''
+    if isinstance(obj, Contribution):
+        sub_id = f':contribution:{obj.id}'
+    elif isinstance(obj, SessionBlock):
+        sub_id = f':block:{obj.id}'
+
+    zoom_id = vc_room.data['zoom_id']
+
+    return f'zoom_meeting:{user_id}@indico:event:{event_id}{sub_id}#{zoom_id}'
```

Comment:
Is `#` really a good idea for something that becomes part of a URL path?

Reply:
Good point. I can just replace it with another colon.